### ISSUE TITLE
services sometimes partially drop off robustirc.net

### ISSUE THREAD

**Issue by stapelberg** [robustirc/robustirc#157]
Last reported 2018-10-06, following a reboot of ex61 at 2018-10-03.

Curiously enough, the HostServ session was still present (but no other session), and the services process was getting new messages from RobustIRC.

Couldn’t reproduce the issue in a local test with restarting either services or RobustIRC processes.

**Comment by stapelberg** [comment]
This kept happening from time to time.

Today I got the chance to look at it in some more detail. I noticed:

* Sometimes the robustirc-bridge process, which the services use to connect to RobustIRC, is stopped first during shutdown. it prints:
  * `Jul 10 08:00:00 ex622 docker[4698]: 2022/07/10 06:00:00 [session 0x40826d777489ce05] Deleting RobustSession. quitmsg="Bridge exiting upon receiving signal (terminated)"`
  * Normally, when anope services shut down, they quit all pseudo users: `:ChanServ QUIT :Shutting down`. This seems to be missing here.
* This shutdown behavior results in the pseudo users like `ChanServ` still being in the RobustIRC network.
* The new services process now starts up and tries to register its `ChanServ` user etc.
  * RobustIRC replies: `:robustirc.net 433 * ChanServ :Nickname is already in use`
  * There is no code in anope services to handle this situation and the services are now in limbo: they are running but effectively unreachable and will not recover from the situation.
  * During the first 10 minutes, when the `ChanServ` ghost user is still in the RobustIRC network, the services can even **act** as e.g. `ChanServ`, making it look like they started up correctly!

I think adding error handling for 433 messages in anope should fix this failure mode quite effectively.
